app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,103 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public void deleteSchedulePlan(StudyIdentifier studyIdentifier, String guid) {
-        checkNotNull(studyIdentifier, "Study identifier is null");
-        checkArgument(StringUtils.isNotBlank(guid), "Plan GUID is blank or null");
+        checkNotNull(studyIdentifier);
+        checkArgument(isNotBlank(guid));
         
         SchedulePlan plan = getSchedulePlan(studyIdentifier, guid);
+        
+        forEachCriteria(plan, scheduleCriteria -> deleteCriteria(scheduleCriteria));
         mapper.delete(plan);
     }
+    
+    private void forEachCriteria(SchedulePlan plan, Function<ScheduleCriteria,Criteria> consumer) {
+        if (plan.getStrategy() instanceof CriteriaScheduleStrategy) {
+            CriteriaScheduleStrategy strategy = (CriteriaScheduleStrategy)plan.getStrategy();
+            for (int i=0; i < strategy.getScheduleCriteria().size(); i++) {
+                ScheduleCriteria scheduleCriteria = strategy.getScheduleCriteria().get(i);
+                
+                // Add the key in order to load
+                scheduleCriteria.getCriteria().setKey(getKey(plan, i));
+
+                Criteria criteria = consumer.apply(scheduleCriteria);
+                
+                // Update the criteria object (except for delete). This may add a new criteria object, 
+                // which will return the plan to the caller with the criteria stubbed out in the JSON.
+                if (criteria != null) {
+                    scheduleCriteria = new ScheduleCriteria(scheduleCriteria.getSchedule(), criteria);
+                    strategy.getScheduleCriteria().set(i, scheduleCriteria);
+                }
+            }
+        }        
+    }
+    
+    private String getKey(SchedulePlan plan, int index) {
+        return "scheduleCriteria:" + plan.getGuid() + ":" + index;
+    }
+    
+    /**
+     * Save the criteria object if it exists. If not, return an empty criteria object which will return the 
+     * criteria stubbed out in the JSON representation of the schedule plan.
+     */
+    private Criteria persistCriteria(ScheduleCriteria scheduleCriteria) {
+        Criteria criteria = scheduleCriteria.getCriteria();
+        if (criteria != null) {
+            criteriaDao.createOrUpdateCriteria(criteria);
+        }
+        return (criteria != null) ? criteria : Criteria.create();
+    }
+
+    /**
+     * Load criteria. If the criteria object doesn't exist, create and return one as part of the schedule plan.
+     */
+    private Criteria loadCriteria(ScheduleCriteria scheduleCriteria) {
+        String key = scheduleCriteria.getCriteria().getKey();
+        Criteria criteria = criteriaDao.getCriteria(key);
+        if (criteria == null) {
+            criteria = Criteria.create();
+            criteria.setKey(key);
+        }
+        return criteria;
+    }
```

Comment:
When will the null case happen? Persist criteria ensures we always create a criteria, and there are no pre-existing CriteriaScheduleStrategies in DDB, so there's no back-compat case to handle.